### PR TITLE
KAFKA-16998 Fix warnings in our Github actions

### PR DESCRIPTION
**Pull request by brandboat**
related to https://issues.apache.org/jira/browse/KAFKA-16998

bump 
- `actions/checkout@v3` to v4
- `actions/setup-python@v3` to v5
- `actions/upload-artifact@v3` to v4
- `actions/stale@v8` to v9
The action above are using nodejs16 (deprecated), updated them to a newer version which use node20. 

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: .github/workflows/docker_build_and_test.yml [hunk 1]
```diff
@@ -32,9 +32,9 @@ jobs:
   build:
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v3
+    - uses: actions/checkout@v4
     - name: Set up Python 3.10
-      uses: actions/setup-python@v3
+      uses: actions/setup-python@v4
```
Comment (chia7712) [inline .github/workflows/docker_build_and_test.yml]: It seems the latest version is `v5`, so should we try it?

**Review by chia7712 — COMMENTED** [review]
@brandboat thanks for patch

File: .github/workflows/docker_build_and_test.yml (hunk 1 above)
Comment (brandboat) [reply .github/workflows/docker_build_and_test.yml]: Sure

**Comment by chia7712** [comment]
@KrishVora01 could you please take a look?

**Comment by VedarthConfluent** [comment]
LGTM, thanks for the PR

**Pull request merged**